src/schedule/SchedulePing.ts

```diff
@@ -12,25 +12,50 @@ enum StartJobsStatus {
 export class SchedulePing {
   private handle?: NodeJS.Timeout;
   private startJobsStatus: StartJobsStatus = StartJobsStatus.notStarted;
+  private readonly maxPingRetries: number;
+  private readonly retryIntervalMs: number;
 
   constructor(
     private readonly schedulesRepository: SchedulesRepository,
     private readonly logger: Logger,
     private readonly interval: number,
     private readonly startAllJobs: () => Promise<void>,
-  ) {}
+    maxPingRetries: number = 1,
+    retryIntervalMs: number = 1_000,
+  ) {
+    this.maxPingRetries = maxPingRetries;
+    this.retryIntervalMs = retryIntervalMs;
+  }
 
   async start(): Promise<void> {
     if (this.handle) {
       return;
     }
     const errorMessage = 'Pinging or cleaning the Schedules repository failed';
     try {
-      await this.checkActiveSchedule();
+      await this.checkActiveScheduleWithRetries(errorMessage);
     } catch (e) {
       this.logger.error(errorMessage, MomoErrorType.internal, this.schedulesRepository.getLogData(), e);
     }
-    this.handle = setSafeInterval(this.checkActiveSchedule.bind(this), this.interval, this.logger, errorMessage);
+    this.handle = setSafeInterval(
+      this.checkActiveScheduleWithRetries.bind(this, errorMessage),
+      this.interval,
+      this.logger,
+      errorMessage,
+    );
+  }
+
+  private async checkActiveScheduleWithRetries(errorMessage: string): Promise<void> {
+    for (let attempt = 1; attempt <= this.maxPingRetries; attempt++) {
```

Comment:
Surely we want the run variable to start at `0`? :D All the programming jokes wouldn't make sense otherwise!